src/Dependencies/CodeAnalysis.Metadata/CustomDebugInfoEncoder.cs

```diff
@@ -143,20 +143,59 @@ public void AddDynamicLocals(IReadOnlyCollection<(string LocalName, byte[] Flags
             AddRecord(
                 CustomDebugInfoKind.DynamicLocals,
                 dynamicLocals,
-                (info, builder) =>
+                (infos, builder) =>
                 {
-                    builder.WriteInt32(dynamicLocals.Count);
+                    builder.WriteInt32(infos.Count);
 
-                    foreach (var (name, flags, count, slotIndex) in dynamicLocals)
+                    foreach (var info in infos)
                     {
-                        Debug.Assert(flags.Length == DynamicAttributeSize);
-                        Debug.Assert(name.Length <= IdentifierSize);
-
-                        builder.WriteBytes(flags);
-                        builder.WriteInt32(count);
-                        builder.WriteInt32(slotIndex);
-                        builder.WriteUTF16(name);
-                        builder.WriteBytes(0, sizeof(char) * (IdentifierSize - name.Length));
+                        Debug.Assert(info.Flags.Length <= DynamicAttributeSize);
+                        Debug.Assert(info.LocalName.Length <= IdentifierSize);
+
+                        builder.WriteBytes(info.Flags);
+                        builder.WriteBytes(0, sizeof(byte) * (DynamicAttributeSize - info.Flags.Length));
```

Comment:
I didn't understand this new output. Looks like we were not padding with zeros. Was that a bug?